includes/functions/shared.php

```diff
@@ -112,6 +112,62 @@ function simcal_get_calendar( $object ) {
 	$objects = \SimpleCalendar\plugin()->objects;
 	return $objects instanceof \SimpleCalendar\Objects ? $objects->get_calendar( $object ) : null;
 }
+/**
+ * Formats strings to ical standard
+ *
+ * @param string $s
+ *
+ * @return string
+ */
+function format_ical_string( $s ) {
+    $r = wordwrap(
+        preg_replace(
+            array( '/,/', '/;/', '/[\r\n]/' ),
+            array( '\,', '\;', '\n' ),
+            $s
+        ), 73, "\n", TRUE
+    );
+
+    // Indent all lines but first:
+    $r = preg_replace( '/\n/', "\n  ", $r );
+
+    return $r;
+}
+
+
+/**
+ * Compose and return a ics feed from the calendar
+ * @param  string|int|object|WP_Post $object
+ * @return void
+ *
+ */
+function simcal_draw_calendar_ics($post) {
+	$calendar = simcal_get_calendar($post);
+	$events = $calendar->events;
+	header('Content-Type:text/calendar');
+	$ics = "BEGIN:VCALENDAR\r\nVERSION:2.0\r\nPRODID:-//Wordpress/Simple-Calendar\r\n";
+
+	foreach($events as $eventarr) {
+		$event = $eventarr[0];
+		if(strpos($ics, $event->uid) == false) {
+			$ics .= "BEGIN:VEVENT\r\n";
+			$ics .="UID:" . $event->uid . "\r\n";
+			$ics .="DTSTAMP:" . $event->start_dt->format('Ymd\THis') . "\r\n";
+			$ics .= "ORGANIZER;CN=" . $event->source . ":MAILTO:".$event->source."\r\n";
+			$ics .= "DTSTART:" . $event->start_dt->format('Ymd\THis') . "\r\n";
+			$ics .= "DTEND:" . $event->end_dt->format('Ymd\THis') . "\r\n";
+			$ics.= "SUMMARY:" . format_ical_string($event->title) . "\r\n";
+			$ics .= "DESCRIPTION:" . format_ical_string($event->description) . "\r\n";
```

Comment:
It would nice to check if the description is present, if not we can skip adding the description.

Reply:
Same for LOCATION and GEO.